### ISSUE TITLE
No reply from ysfreflector

### ISSUE THREAD

**Issue by pschroth** [g4klx/YSFClients#18]
The ysfreflector is not responding to pings from the gateway.
It sends a empty packet back.

Magically it starts to respond in the right way, but after restart of the process, the ping replys were empty again.

Kim DG9VH gave me some advice to pinpoint the problem. Sadly there was no solution. In the Yahoo group there are at least two more with the same problem.

Thanks Philip

Ysfgateway.ini

[General]
Callsign=PA3DFN
Suffix=RPT
# Suffix=ND
RptAddress=127.0.0.1
RptPort=3200
LocalAddress=127.0.0.1
LocalPort=4200
Daemon=0

[Info]
RXFrequency=435000000
TXFrequency=435000000
Power=1
Latitude=0.0
Longitude=0.0
Height=0
Name=Test
Description=Multi-Mode Repeater

[Log]
# Logging levels, 0=No logging
DisplayLevel=1
FileLevel=1
FilePath=.
FileRoot=YSFGateway

[aprs.fi]
Enable=0
# Server=noam.aprs2.net
Server=euro.aprs2.net
Port=14580
Password=22185

[Network]
Enable=1
DataPort=24000
StatusPort=24001
Hosts=./YSFHosts.txt
Debug=1

Ysfreflector.ini
[General]
Daemon=1

[Info]
Name=NL YSF444
Description=Dutch Refl.

[Log]
# Logging levels, 0=No logging
DisplayLevel=0
FileLevel=1
FilePath=/var/log
FileRoot=YSFReflector

[Network]
Port=42000
Debug=1

**Comment by dg9vh** [comment]
For all that have the same problem:
New YSFReflectors (or reflectors with changed configuration) have to be "waked up" by following procedure:

1. Configure MMDVMHost to directly connect to the Reflector using following configuration (depending on the reflectors address):
   [System Fusion Network]

   Enable=1

   LocalAddress=127.0.0.1

   LocalPort=/dev/null

   GwyAddress=your.hostname.net

   GwyPort=42000

   Debug=1
2. Start MMDVMHost and make a YSF-Transmission
3. Check with `echo -n "YSFS" | nc -u your.hostname.net 42000` if the reflector is responding
4. Configure your MMDVMHost to use your YSFGateway again

For final: This is a serious issue to be fixed!

**Comment by dg9vh** [comment]
@pschroth @g4klx I think I fixed this issue. After testing around today with serveral instances of YSFReflector on serveral systems they showed all the same issue: After startup they replied weird to YSFS-request so long as no Gateway/MMDVMHost was connected. After this connect they replied with the count 1 and then after disconnecting with 0. Looking to the code brought me to the point that the count was not initialized with 0 at startup so it happens that so long no endpoint was connected, weird replies to the YSFS-request followed.

Tested with my modified version it was reliable answered with 0 if no one was connected.
